sound/soc/sof/topology.c

```diff
@@ -1898,6 +1898,12 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	case snd_soc_dapm_siggen:
 		ret = sof_widget_load_siggen(scomp, index, swidget, tw, &reply);
 		break;
+	/*
+	 * snd_soc_dapm_sink can treated as a special processing type widget.
+	 * So, re-use the same ipc structure and tokens as the effect widgets.
```

Comment:
the explanation needs to be clarified. We could state the same thing for the siggen just above, why can't it be treated as an effect?